app/controllers/api/v2/search_controller.rb

```diff
@@ -24,6 +28,49 @@ def matching_suggestions
 
         []
       end
+
+      def log_search_results(query, results)
+        results_type = results[:data][:type]
+        attributes = results[:data][:attributes]
+
+        commodity_score = attributes[:goods_nomenclature_match]['commodities'].first&.dig('_score') || 0
+        chapter_score = attributes[:goods_nomenclature_match]['chapters'].first&.dig('_score') || 0
+        heading_score = attributes[:goods_nomenclature_match]['headings'].first&.dig('_score') || 0
+        section_score = attributes[:goods_nomenclature_match]['sections'].first&.dig('_score') || 0
+
+        ref_commodity_score = attributes[:reference_match]['commodities'].first&.dig('_score') || 0
+        ref_chapter_score = attributes[:reference_match]['chapters'].first&.dig('_score') || 0
+        ref_heading_score = attributes[:reference_match]['headings'].first&.dig('_score') || 0
+        ref_section_score = attributes[:reference_match]['sections'].first&.dig('_score') || 0
+
+        # Find the maximum score
+        max_score = [
+          commodity_score,
+          chapter_score,
+          heading_score,
+          section_score,
+          ref_commodity_score,
+          ref_chapter_score,
+          ref_heading_score,
+          ref_section_score,
+        ].max
+
+        results_count = attributes[:goods_nomenclature_match]['chapters'].size + attributes[:reference_match]['chapters'].size + attributes[:goods_nomenclature_match]['commodities'].size + attributes[:reference_match]['commodities'].size + attributes[:goods_nomenclature_match]['headings'].size + attributes[:reference_match]['headings'].size + attributes[:goods_nomenclature_match]['sections'].size + attributes[:reference_match]['sections'].size
+
+        results_zero = results_count.zero?
+        query_length = query.present? ? query.length : 0
+        Rails.logger.info "Search Request : Search Query [#{query}] | Query Length [#{query_length}] | "\
```

Comment:
Can we spit this out as key=value rather than just a pipe concatenated string.  key=value is usually much more parseable by logging tools.

Reply:
Did this change now. Logs will be seen as 
 [2025-03-10T17:20:09.001292 #71927]  INFO -- : Search Request : search_query=cherry, query_length=6, result_type=fuzzy_search, top_result_score=11.181876, result_count=134, result_zero=false
 
 [2025-03-10T17:20:59.806901 #71926]  INFO -- : Search Suggestion Request : search_query=horses, query_length=6, result_count=1, result_zero=false

Reply:
@neilmiddleton Is this format good?